src/endian.c

```diff
@@ -0,0 +1,36 @@
+#include "endian.h"
+
+// See https://github.com/nodejs/node/blob/master/src/util.h
+const union {
+  uint8_t u8[2];
+  uint16_t u16;
+} byte_order = {{1, 0}};
+
+udx__endianness_t
+udx__endianness () {
+  return byte_order.u16 == 1 ? UDX_LE : UDX_BE;
+}
+
+bool
+udx__is_le () {
+  return udx__endianness() == UDX_LE;
+}
+
+bool
+udx__is_be () {
+  return udx__endianness() == UDX_BE;
+}
+
+uint32_t
+udx__swap_uint32 (uint32_t x) {
+  return ((x & 0x000000ff) << 24) |
+         ((x & 0x0000ff00) << 8) |
+         ((x & 0x00ff0000) >> 8) |
+         ((x & 0xff000000) >> 24);
+}
+
+uint32_t
+udx__swap_uint32_if_be (uint32_t n) {
+  if (udx__is_le()) return n;
+  return udx__swap_uint32(n);
+}
```

Comment:
With `-O2` this compiles to the identity function on LE systems.

Reply:
Nice, how do you see that? (i'm a noob and could use that skill)

Reply:
I compiled to assembly:

```sh
cc -S -O2 endian.c
```

Reply:
The assembly output will be in `endian.s`.

Reply:
But what do you use to inspect that as a human?

Reply:
I personally just read it, disregarding most of the boilerplate 👀